processor/src/main/java/com/infinum/jsonapix/processor/JsonApiProcessor.kt

```diff
@@ -244,71 +243,136 @@ public class JsonApiProcessor : AbstractProcessor() {
         listFileSpec.writeTo(File(kaptKotlinGeneratedDir))
     }
 
-    private fun processErrorAnnotation(roundEnv: RoundEnvironment?) {
-        roundEnv?.getElementsAnnotatedWith(JsonApiXError::class.java).orEmpty().forEach {
-            val type = it.getAnnotationParameterValue<JsonApiXError, String> { type }
-            val className = ClassName(
-                processingEnv.elementUtils.getPackageOf(it).toString(),
-                it.simpleName.toString()
-            )
-            customErrors[type] = className
+    private fun RoundEnvironment.processErrorAnnotation() {
+        val repeatedAnnotations = this.getElementsAnnotatedWith(JsonApiXErrorList::class.java)
+        val singleAnnotations = this.getElementsAnnotatedWith(JsonApiXError::class.java)
+
+        repeatedAnnotations?.forEach { element ->
+            val types = element.getAnnotationParameterValues<JsonApiXError, List<String>> {
+                this.map { annotation -> annotation.type }
+            }
+            types.forEach { type ->
+                storeCustomError(element = element, type = type)
+            }
         }
+
+        singleAnnotations?.forEach { element ->
+            val type = element.getAnnotationParameterValue<JsonApiXError, String> { type }
+            storeCustomError(element = element, type = type)
+        }
+
```

Comment:
For all three annotations, we have to "fetch" both the single annotations(`JsonApiXError`) and list annotations (`JsonApiXErrorList`). For example, lets say we have a class:

```
@Serializable
@JsonApiXError("person")
data class PersonalError(val desc: String) : Error

@Serializable
@JsonApiXError("dog")
@JsonApiXError("company")
data class ReusableError(val desc: String) : Error
```
The `PersonalError` will get returned with the call to `getElementsAnnotatedWith(JsonApiXError::class.java)`.

The `ReusableError` will get returned with the call to `getElementsAnnotatedWith(JsonApiXErrorList::class.java)`

So we in all casses we have to handle both types of the annotations. For the user nothing chagnes, they are only aware and use the single(`JsonApiXError`) annotaton.